examples/gno.land/p/demo/timelock/timelock.gno

```diff
@@ -0,0 +1,150 @@
+// Package timelock provides a library for scheduling, cancelling, and
+// executing time-locked operations in Gno. It ensures that
+// operations are only carried out after a specified delay and offers
+// mechanisms for managing and verifying the status of these operations.
+// This package leverages an AVL tree for efficient management of timestamps
+// and integrates role-based access control for administrative tasks.
+//
+// Example Usage:
+//
+//  import "gno.land/p/demo/timelock"
+//  import "gno.land/p/demo/accesscontrol"
+//
+//  Initialize timelock utility with an AVL tree and access control.
+//  timestamps := avl.NewTree()
+//  adminRole := accesscontrol.NewRoleData("admin", std.Address("admin-address"))
+//  timelockUtil := timelock.NewTimelockUtil(timestamps, adminRole)
+//
+//  Schedule an operation with a delay of 60 seconds and a minimum delay of 30 seconds.
+//  id := seqid.ID()
+//  timelockUtil.Schedule(id, 60, 30)
+//
+//  Check if an operation is pending.
+//  isPending := timelockUtil.IsOperationPending(id)
+//
+//  Execute the operation when it is ready.
+//  if timelockUtil.IsOperationReady(id) {
+//      timelockUtil.Execute(id)
+//  }
+//
+//  Update the minimum delay for future operations.
+//  var minDelay int64 = 30
+//  timelockUtil.UpdateDelay(&minDelay, 45)
+
+package timelock // import "gno.land/p/demo/timelock"
+
+import (
+    "strconv"
+    "std"
+    "gno.land/p/demo/avl"
+    "gno.land/p/demo/accesscontrol"
+    "gno.land/p/demo/seqid"
+    "time"
+)
+
+// Marks operations completed
+const DONE_TIMESTAMP = 1
+
+// Represents the status of a planned operation
+type OperationState int
+
+const (
+    Unset OperationState = iota
+    Waiting
+    Ready
+    Done
+)
+
+// TimelockUtil stores the necessary parameters for the timelock operations
+type TimelockUtil struct {
+    timestamps    *avl.Tree
```

Comment:
Right now we use avl trees as a stand in for maps due to performance reasons.  In this case, the value of `timestamps` is provided by the caller of `NewTimelockUtil`.  I was considering what would happen if we some day come up with a better map-like implementation; I think there would have to be a new package with the type of `TimelockUtil.timestamps` changed to that new type.

However, if `timestamps` is an interface with only the methods it needs, it could serve to future proof it a bit.  I'm suggesting something like:
```go
type MapStore interface {
  Set...
  Get...
  Has...
  etc.
}
```
This is not a requirement of course, but think about it.